modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/GeneratedMethodComponentExecutor.java

```diff
@@ -164,7 +166,7 @@ public void setMuleContext(MuleContext context) {
         }
         return resolvedParams;
       } finally {
-        setContextClassLoader(thread, extensionClassLoader, currentClassLoader);
+        setContextClassLoader(thread, compositeClassLoader, currentClassLoader);
```

Comment:
We would need to review this with the improvement made by Mariano Gonzalez for performance as switching class loaders in TCCL was a "problem". On this case we cannot 'reuse' the composite as it will be created with the current TCCL which could be a domain or application.